### PR TITLE
Read views - box.read_view.list

### LINKED ISSUE #3337: Document box.read_view.list
The new function returns an array of all active database read views.
It includes both read views created for system purposes (e.g. to make a
checkpoint or join a new replica) and read views created by application
code (this feature is available only in Tarantool Enterprise Edition,
see https://github.com/tarantool/enterprise_doc/issues/194).

Each read view is represented by a table with the following fields:
 - `id` - unique read view identifier.
 - `name` - read view name.
 - `is_system` - true if the read view is used for system purposes.
 - `timestamp` - `fiber.clock()` when the read view was opened.
 - `vclock` - `box.info.vclock` when the read view was opened.
 - `signature` - `box.info.signature` when the read view was opened.
 - `status` - 'open' or 'closed'.

Read views created by application code also have the 'space' field,
which lists all spaces available in the read view, and may be used just
like a read view object returned by `box.read_view.open()`.

The array is sorted by read view id. Since read view ids grow
monotonically, this means that the most recent read view goes last.

Example:

```
tarantool> box.read_view.list()
---
- - timestamp: 4057333.4969064
    signature: 3
    is_system: true
    status: open
    vclock: {1: 3}
    name: join
    id: 3
  - timestamp: 4057357.0874192
    signature: 6
    is_system: true
    status: open
    vclock: {1: 6}
    name: checkpoint
    id: 4
...
```
Requested by @locker in https://github.com/tarantool/tarantool/commit/de5f19e9c6b38518ef372ce41de6de716f43bf93.

### PR DESCRIPTION
**Pull request by andreyaksenov**
Added [box.read_view.list()](https://docs.d.tarantool.io/en/doc/2.11-read-views/reference/reference_lua/box_read_view/list/) to reference.

### CODE REVIEW

File: doc/reference/reference_lua/box_read_view/list.rst [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    This array might include the following read view types:
+
+    *   `read views <https://www.tarantool.io/en/enterprise_doc/read_views/>`_ created by application code (available in Tarantool Enterprise only)
+
+    *   system read views (used, for example, to make a :ref:`checkpoint <book_cfg_checkpoint_daemon>`
+        or join a new :ref:`replica <replication-architecture>`)
+
+    Read views created by application code also have the ``space`` field,
+    which lists all spaces available in a read view, and may be used like
+    a read view object returned by ``box.read_view.open()``.
```
Comment (locker) [inline doc/reference/reference_lua/box_read_view/list.rst]: It also contains read views created with the "raw" C API (`box_raw_read_view_new()`). Those read views don't have a space list (like system read views, they can't be accessed from Lua).

**Review by locker — APPROVED** [review]

File: doc/reference/reference_lua/box_read_view/list.rst (hunk 1 above)
Comment (andreyaksenov) [reply doc/reference/reference_lua/box_read_view/list.rst]: thanks, will fix

File: doc/reference/reference_lua/box_read_view/list.rst [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
```
Comment (xuniq) [inline doc/reference/reference_lua/box_read_view/list.rst]:
```suggestion
    Read views created by application code also have the ``space`` field. The field lists all spaces available in a read view, and may be used like
    a read view object returned by ``box.read_view.open()``.
```

File: doc/reference/reference_lua/box_read_view.rst [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+    ..  list-table::
+        :widths: 25 75
+        :header-rows: 1
+
+        *   - Name
+            - Use
+
+        *  - :doc:`./box_read_view/list`
+           - Returns an array of all active database read views.
```
Comment (xuniq) [inline doc/reference/reference_lua/box_read_view.rst]:
```suggestion
           - Return an array of all active database read views.
```

From the [guidelines](https://www.tarantool.io/en/doc/latest/contributing/docs/api/#language-of-the-general-description):

> Use one of the two options:
> * Start with a verb in the imperative mood. Example: _Create a fiber_.
> * Start with a noun. Example: _The directory where memtx stores snapshot files._

Example: 
https://www.tarantool.io/en/doc/latest/reference/reference_lua/box_info/

File: doc/reference/reference_lua/box_read_view/list.rst [hunk 4]
```diff
@@ -0,0 +1,48 @@
+.. _reference_lua-box_read_view_list:
+
+===============================================================================
+box.read_view.list()
+===============================================================================
+
+.. function:: read_view.list()
+
+    Returns an array of all active database read views.
```
Comment (xuniq) [inline doc/reference/reference_lua/box_read_view/list.rst]:
```suggestion
    Return an array of all active database read views.
```

**Review by xuniq — APPROVED** [review]
Looks good, added several suggestions.

**Pull request merged**